discovery-provider/src/api/v1/users.py

```diff
@@ -625,3 +627,48 @@ def get(self):
         top_users = get_top_genre_users(get_top_genre_users_args)
         users = list(map(extend_user, top_users['users']))
         return success_response(users)
+
+associated_wallet_route_parser = reqparse.RequestParser()
+associated_wallet_route_parser.add_argument('id', required=True)
+associated_wallet_response = make_response("associated_wallets_response", ns, fields.Nested(associated_wallets))
+@ns.route("/associated_wallets")
+class UserIdByAssociatedWallet(Resource):
+    @ns.expect(associated_wallet_route_parser)
+    @ns.doc(
+        id="""Get the User's id by associated wallet""",
+        params={'id': 'Encoded User ID'},
+        responses={
+            200: 'Success',
+            400: 'Bad request',
+            500: 'Server error'
+        }
+    )
+    @ns.marshal_with(associated_wallet_response)
+    @cache(ttl_sec=10)
+    def get(self):
+        args = associated_wallet_route_parser.parse_args()
+        user_id = decode_with_abort(args.get('id'), ns)
+        wallets = get_associated_user_wallet({ "user_id": user_id})
+        return success_response({ "wallets": wallets })
+
+user_associated_wallet_route_parser = reqparse.RequestParser()
+user_associated_wallet_route_parser.add_argument('associated_wallet', required=True)
+user_associated_wallet_response = make_response("user_associated_wallet_response", ns, fields.Nested(encoded_user_id))
+@ns.route("/id")
+class AssociatedWalletByUserId(Resource):
+    @ns.expect(user_associated_wallet_route_parser)
+    @ns.doc(
+        id="""Get the User's associated wallets""",
+        params={'associated_wallet': 'Wallet address'},
+        responses={
+            200: 'Success',
+            400: 'Bad request',
+            500: 'Server error'
+        }
+    )
+    @ns.marshal_with(user_associated_wallet_response)
+    @cache(ttl_sec=1)
```

Comment:
why the 1s here and 10s there?

Reply:
ahh, yes wasn't sure what values to put, but will make them both the same.